packages/engine/src/lib/helper/piece-helper.ts

```diff
@@ -216,7 +217,10 @@ export const pieceHelper = {
                 disabled: true,
                 options: [],
                 placeholder: "Throws an error, reconnect or refresh the page"
-            } as DropdownState<unknown>;
+            } as DropdownState<unknown> & {
```

Comment:
casting shouldn't happen, can you handle the static case differently. that would make it work.